src/urllib3/connectionpool.py

```diff
@@ -744,17 +744,18 @@ def urlopen(  # type: ignore
             # Discard the connection for these exceptions. It will be
             # replaced during the next _get_conn() call.
             clean_exit = False
+            new_e: Exception = e
```

Comment:
@sethmlarson @hramezani What do you think about that `new_e` name? I had to use another name because mypy does not support assignments that change the type, but I did not find a good name. Now we have `e`, `new_e` and `err`.

Reply:
Yes, you are right. I think we have to do it like this.